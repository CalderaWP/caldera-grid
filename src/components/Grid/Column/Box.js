import React from 'react';
import PropTypes from 'prop-types';
export  const Box = (props) => {
    return (
        <div>
            {props.content}
        </div>
    )
};

Box.propTypes = {
    content: PropTypes.string,
};