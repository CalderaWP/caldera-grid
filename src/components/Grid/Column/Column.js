import React from 'react';
import PropTypes from 'prop-types';


export const chooseColumnBox = (boxMode,boxes,defaultMode) => {
    if( boxes.hasOwnProperty(boxMode)){
        return boxes[boxMode];
    }else if( boxes.hasOwnProperty(defaultMode)){
        return boxes[defaultMode]
    }
    return boxes[Object.keys(boxes)[0]];
};


export const Column = (props) => {
    const defaultMode = props.hasOwnProperty('defaultMode') ? props.defaultMode : 'previewMode';
    return (
        <div>
            {chooseColumnBox(props.boxMode,props.boxes,defaultMode)}
        </div>
    )
};

Column.propTypes = {
    content: PropTypes.string,
    boxes: PropTypes.object,
    boxMode: PropTypes.string,
    defaultMode: PropTypes.string
};