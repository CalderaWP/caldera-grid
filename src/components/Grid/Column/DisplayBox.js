import React from 'react';
import PropTypes from 'prop-types';
export  const DisplayBox = (props) => {
    return (
        <div>
            {props.content}
        </div>
    )
};

DisplayBox.propTypes = {
    content: PropTypes.string,
};