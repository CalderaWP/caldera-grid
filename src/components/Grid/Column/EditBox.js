import React from 'react';
import PropTypes from 'prop-types';
export  const EditBox = (props) => {
    return (
        <div>
            <input type={'text'} value={props.content}/>
        </div>
    )
};

EditBox.propTypes = {
    content: PropTypes.string.isRequired,
};