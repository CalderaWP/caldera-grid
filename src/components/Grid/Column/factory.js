import {EditBox} from "./EditBox";
import {DisplayBox} from "./DisplayBox";

export const defaultMode = 'displayMode';
export const prepareColumn = (column) => {
    const boxProps = {
        content: column.content
    };
    column.boxes = {
        editMode: EditBox(boxProps),
        displayMode:DisplayBox(boxProps)
    };

    return column;
};
export const prepareRow = (row) => {
    Object.keys(row.columns).map( (columnId) => {
        row.columns[columnId] = prepareColumn(row.columns[columnId])
    });
    row.boxMode = row.hasOwnProperty('mode' ) ? row.mode : defaultMode;
    row.defaultMode = defaultMode;
    return row;
};