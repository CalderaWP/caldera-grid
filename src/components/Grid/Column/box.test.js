import React from 'react';
import renderer from 'react-test-renderer';
import {chooseColumnBox, Column} from "./Column";
import {Box} from "./Box";
import {DisplayBox} from "./DisplayBox";
import {EditBox} from "./EditBox";

describe('Chooses boxes correctly', () => {

    it( 'chooses the right existent box',  () => {
        const props = {content:'hi'};
        const boxes = {
            editMode: EditBox(props),
            displayMode: DisplayBox(props)
        };

        let chosen = chooseColumnBox('editMode',boxes,'insaneMode' );
        expect(chosen).toEqual(boxes['editMode'])

    });

    it( 'chooses the default box when asked for non-existent box ',  () => {
        const props = {content:'hi'};
        const boxes = {
            otherMode: Box(props),
            editMode: EditBox(props),
            displayMode: DisplayBox(props)
        };

        let chosen = chooseColumnBox('insaneMode',boxes,'editMode' );
        expect(chosen).toEqual(boxes['editMode'])

    });

    it( 'chooses the first box when asked for non-existent box and default box does not exist either',  () => {
        const props = {content:'hi'};
        const boxes = {
            editMode: EditBox(props),
            displayMode: DisplayBox(props)
        };

        let chosen = chooseColumnBox('insaneMode',boxes,'crazyMode' );
        expect(chosen).toEqual(boxes['editMode'])

    });

});


describe('Outputs the right box', () => {

    it( 'Renders the right existent box',  () => {
        const boxProps = {content:'hi'};
        const boxes = {
            editMode: EditBox(boxProps),
            displayMode: DisplayBox(boxProps)
        };

        const component = renderer.create(
            <Column
                content={'Hi Roy'}
                boxes={boxes}
                boxMode={'editMode'}
                defaultMode={'fakeMode'}
            />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    });

    it( 'Renders the default box when asked for non-existent box ',  () => {
        const props = {content:'hi'};
        const boxes = {
            otherMode: Box(props),
            editMode: EditBox(props),
            displayMode: DisplayBox(props)
        };

        const component = renderer.create(
            <Column
                content={'Hi Roy'}
                boxes={boxes}
                boxMode={'depecheMode'}
                defaultMode={'editMode'}
            />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    });

    it( 'Render the first box when asked for non-existent box and default box does not exist either',  () => {
        const props = {content:'hi'};
        const boxes = {
            editMode: EditBox(props),
            displayMode: DisplayBox(props)
        };

        const component = renderer.create(
            <Column
                content={'Hi Roy'}
                boxes={boxes}
                boxMode={'depecheMode'}
                defaultMode={'fakeMode'}
            />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    });

});

