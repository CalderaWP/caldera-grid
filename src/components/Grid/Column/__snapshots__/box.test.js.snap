// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Outputs the right box Render the first box when asked for non-existent box and default box does not exist either 1`] = `
<div>
  <div>
    <input
      type="text"
      value="hi"
    />
  </div>
</div>
`;

exports[`Outputs the right box Renders the default box when asked for non-existent box  1`] = `
<div>
  <div>
    <input
      type="text"
      value="hi"
    />
  </div>
</div>
`;

exports[`Outputs the right box Renders the right existent box 1`] = `
<div>
  <div>
    <input
      type="text"
      value="hi"
    />
  </div>
</div>
`;
